src/Illuminate/Database/Eloquent/Relations/Relation.php

```diff
@@ -273,6 +280,21 @@ public function wrap($value)
     }
 
     /**
+     * Set the morph map for polymorphic relations.
+     * 
+     * @param  array|null $map
+     * @return array
+     */
+    public static function morphMap($map = null)
+    {
+        if (is_null($map)) {
+            return static::$morphMap;
+        }
+
+        static::$morphMap = $map;
```

Comment:
> Could you give an example @crynobone, not sure I understand the case.

So if a few packages build a morph class, the user need to explicitly set the `Relation::morphMap()` inside the `AppServiceProvider` or this doesn't work. Instead having the freedom to do it on the package service providers.